src/Raven.Server/Documents/PeriodicBackup/BackupTask.cs

```diff
@@ -644,6 +654,43 @@ private class InternalBackupResult
             return internalBackupResult;
         }
 
+        private void DeleteFile(string path)
+        {
+            try
+            {
+                IOExtensions.DeleteFile(path);
+            }
+            catch (Exception e)
+            {
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"Failed to delete file: {path}", e);
+            }
+        }
+
+        private void ValidateFreeSpaceForSnapshot(string filePath)
+        {
+            long totalUsedSpace = 0;
+            foreach (var mountPointUsage in _database.GetMountPointsUsage(includeTempBuffers: false))
+            {
+                totalUsedSpace += mountPointUsage.UsedSpace;
+            }
+
+            var directoryPath = Path.GetDirectoryName(filePath);
+            var destinationDriveInfo = DiskSpaceChecker.GetDiskSpaceInfo(directoryPath);
+            if (destinationDriveInfo == null)
+            {
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"Couldn't find the disk space info for path: {filePath}");
+
+                return;
+            }
+
+            var desiredFreeSpace = Size.Min(new Size(512, SizeUnit.Megabytes), destinationDriveInfo.TotalSize * 0.01) + new Size(totalUsedSpace, SizeUnit.Bytes);
```

Comment:
This actually requires to have 512 MB of free space regardless the backup size. Why?

Reply:
This is the same logic that we use for database restore

Reply:
Okay. That might make sense. We won't consume free space completely this way. Do you think we should extract this logic to some helper method and used it here and during restore, so it will be consistent in both?

Reply:
done